Source/Scene/ShadowMap.js

```diff
@@ -1248,6 +1258,59 @@ define([
         }
     };
 
+    ShadowMap.prototype.combineUniforms = function(uniforms, isTerrain) {
+        var isPointLight = this.isPointLight;
+        var bias = isPointLight ? this._pointBias : (isTerrain ? this._terrainBias : this._primitiveBias);
+
+        var that = this;
+        var mapUniforms = {
```

Comment:
Can any of these be packed into fewer uniforms, e.g., combine light direction/position?